src/parsers/manifest/dash/indexes/base.ts

```diff
@@ -141,18 +143,21 @@ export default class BaseRepresentationIndex implements IRepresentationIndex {
    * @param {Object} index
    * @param {Object} context
    */
-  constructor(index : IBaseIndexIndexArgument, context : IBaseIndexContextArgument) {
+  constructor(index : IBaseIndexIndexArgument,
+              context : IBaseIndexContextArgument,
+              isContentFragmented: boolean) {
     const { periodStart,
             periodEnd,
             representationBaseURL,
             representationId,
             representationBitrate } = context;
     const { timescale } = index;
-
+    this._isContentFragmented = isContentFragmented;
```

Comment:
I'm not sure I follow here...
 - Either we consider "non-fragmented" as segmentBase with one element in their timeline and no init data
 - Either it's not a SegmentBase

Reply:
Well, if we do not have any of indications that it may be a Template or List Representation, it may be either a unique file, that do not have an init segment, or a segmented file were init segment and index are inside the file. In both cases, i consider it to be "segment base".

Reply:
So if it's not fragmented, it has no init data and one element in the timeline no?
Should be as simple as that.

If it is fragmented and we do not know where the index is, then it is the same thing than regular SegmentBase minus the `indexRange` information.

If it is fragmented and we do not know where the initialization data is, then it is the same thing than regular SegmentBase minus the `initialization.range` information.

In any case I don't see the point of that variable.

Reply:
I removed isContentFragment.
If SegmentBase and no init range / index range, we try to request first bytes. If we can't find init segment or index, just add one unique segment in base index.